core/src/main/kotlin/com/malinskiy/marathon/execution/progress/ProgressReporter.kt

```diff
@@ -1,82 +1,25 @@
+import com.malinskiy.marathon.test.TestBatch
 import com.malinskiy.marathon.test.toTestName
-import java.util.concurrent.ConcurrentHashMap
-import kotlin.math.roundToInt
 
-const val HUNDRED_PERCENT_IN_FLOAT: Float = 100.0f
-
-class ProgressReporter(private val configuration: Configuration) {
-    private val reporters = ConcurrentHashMap<DevicePoolId, PoolProgressTracker>()
-
-    private inline fun <T> execute(poolId: DevicePoolId, f: (PoolProgressTracker) -> T): T {
-        val reporter = reporters[poolId] ?: PoolProgressTracker(configuration)
-        val result = f(reporter)
-        reporters[poolId] = reporter
-        return result
-    }
-
-    private fun toPercent(float: Float): String {
-        val percent = (float * HUNDRED_PERCENT_IN_FLOAT).roundToInt()
-        val format = "%02d%%"
-        return String.format(format, percent)
-    }
-
-    fun testStarted(poolId: DevicePoolId, device: DeviceInfo, test: Test) {
-        execute(poolId) { it.testStarted(test) }
-        println("${toPercent(progress(poolId))} | [${poolId.name}]-[${device.serialNumber}] ${test.toTestName()} started")
-    }
-
-    fun testFailed(poolId: DevicePoolId, device: DeviceInfo, test: Test) {
-        execute(poolId) { it.testFailed(test) }
-        println("${toPercent(progress(poolId))} | [${poolId.name}]-[${device.serialNumber}] ${test.toTestName()} failed")
-    }
-
-    fun testPassed(poolId: DevicePoolId, device: DeviceInfo, test: Test) {
-        execute(poolId) { it.testPassed(test) }
-        println("${toPercent(progress(poolId))} | [${poolId.name}]-[${device.serialNumber}] ${test.toTestName()} ended")
-    }
-
-    fun testIgnored(poolId: DevicePoolId, device: DeviceInfo, test: Test) {
-        execute(poolId) { it.testIgnored(test) }
-        println("${toPercent(progress(poolId))} | [${poolId.name}]-[${device.serialNumber}] ${test.toTestName()} ignored")
-    }
-
-    fun aggregateResult(): Boolean {
-        return reporters.isNotEmpty() && reporters.values.all {
-            it.aggregateResult()
-        }
-    }
-
-    fun testCountExpectation(poolId: DevicePoolId, size: Int) {
-        execute(poolId) { it.testCountExpectation(size) }
-    }
-
-    fun removeTests(poolId: DevicePoolId, count: Int) {
-        execute(poolId) { it.removeTests(count) }
-    }
-
-    fun addTestDiscoveredDuringRuntime(poolId: DevicePoolId, test: Test) {
-        execute(poolId) { it.addTestDiscoveredDuringRuntime(test) }
+class ProgressReporter(private val batch: TestBatch, private val poolId: DevicePoolId, private val device: DeviceInfo) {
+    fun testStarted(test: Test) {
+        println("${batch.id} | [${poolId.name}]-[${device.serialNumber}] ${test.toTestName()} started")
     }
 
-    fun addRetries(poolId: DevicePoolId, count: Int) {
-        execute(poolId) { it.addTestRetries(count) }
+    fun testFailed(test: Test) {
+        println("${batch.id} | [${poolId.name}]-[${device.serialNumber}] ${test.toTestName()} failed")
```

Comment:
Same as above. this is the old mechanism to show progress in stdout. I've removed it since most of the info should come from poolprogressaccumulator now